src/compile.rs

```diff
@@ -1630,18 +1605,38 @@ fn do_rules_pass<'a>(
         None => BlockType::Global,
     };
 
-    let mut local_context = BlockContext::new(block_type, types, parent_type);
+    let mut local_context = BlockContext::new(block_type, parent_type, parent_context);
     local_context.insert_function_args(&func_args);
 
     for e in exprs {
         match e {
+            Expression::Stmt(Statement::LetBinding(l)) => {
+                // Need to handle this special case here, otherwise ValidatedStatement::new()
+                // confuses the borrow checker because it might mutate local_context, or return
+                // data that references the context
+                if parent_type.is_some() {
+                    match local_context.insert_from_argument(
+                        &l.name,
+                        &l.value,
+                        class_perms,
+                        types,
+                        file,
+                    ) {
+                        Ok(()) => (),
+                        Err(e) => errors.append(e),
+                    }
+                }
+                // TODO: The note in ValidatedStatement::new() says "TODO: This is where local
+                // scope let bindings should happen".  Isn't that what we just did above?  Is there
+                // still a TODO here?
```

Comment:
I don't think so.  The logic that was in ValidatedStatement was moved here, and the other logic added in this PR seems to be handling the let statements correctly.